File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/TurnosMedicos.java
package com.crui.patterns.examples.turnos_medicos.after;

public class TurnosMedicos {

  public static void main(String[] args) {
    System.out.println();
    System.out.println("Turnos Medicos");
    System.out.println("=============");
    System.out.println();

    Database database = Database.getInstance();

    Paciente paciente = new Paciente("Ignacio Segovia", "OSDE", "isegovia@gmail.com");
    String especialidad = "Cardiología";
    Doctor doctor = database.getDoctor(especialidad);

    if (doctor == null) {
      System.out.println("No se encontró el doctor para la especialidad: " + especialidad);
      return;
    }

    Turno turno = TurnoBuilder.nuevo()
        .paciente(paciente)
        .doctor(doctor)
        .fechaYHora("2025-01-01 10:00") // opcional
        // .precio(12345) // opcional
        .construir();

    System.out.println(turno);

    turno.setFechaYHora("2025-01-01 11:00");
    System.out.println();
  }
}

File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/TurnosMedicos.py
from Paciente import Paciente
from Doctor import Doctor
from Database import Database
from Turno import Turno

class TurnosMedicos:
    @staticmethod
    def main():
        print()
        print("Turnos Medicos")
        print("=============")
        print()

        database: Database = Database.getInstance()

        paciente: Paciente = Paciente("Ignacio Segovia", "OSDE", "isegovia@gmail.com")
        especialidad: str = "Cardiología"
        doctor: Doctor = database.getDoctor(especialidad)

        if doctor is None:
            print("No se encontró el doctor para la especialidad: " + especialidad)
            exit

        # Precio base en base a la especialidad
        

        # Descuento en base a la obra social y la especialidad
        precio = Turno.getTotalPrecio()

        # Nuevo turno
        turno: Turno = Turno(paciente, doctor, "2025-01-01 10:00", precio)
        print(turno)

        # Cambio de turno
        turno.setFechaYHora("2025-01-01 11:00")
        print()

if __name__ == "__main__":
    TurnosMedicos.main()